src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
+		
+		// For cases where this the first Release that a RlE has appeared (ie. fetching
+		// for the RlE in the previous DB returned null).
 
-		}
-		catch (SQLException e)
+		// As before, iterate through all RlEs.
+		for (GKInstance newRLE : newRLEs)
 		{
-			e.printStackTrace();
+			// Get non-Reactome IEs from the revised, authored, and reviewed attributes.
+			List<GKInstance> newRLEInstanceEdits = getNonReactomeEventIEs(newRLE);
+
+			// Look through all parents of the RlE, trying to find the new instance edit in
+			// any of their revised, authored, or reviewed attributes.
+			List<GKInstance> parentPathways = (List<GKInstance>) newRLE.getReferers(ReactomeJavaConstants.hasEvent);
+			for (GKInstance parentPathway : parentPathways)
+			{
+				// Get all non-Reactome instance edits in the revised, authored, and reviewed
+				// attributes of the Pathway.
+				List<GKInstance> parentPathwayIEs = getNonReactomeEventIEs(parentPathway);
+
+				boolean hasRLEInstanceEdit = false;
+
+				// Iterate through all new instance edits, trying to find a Pathway containing
+				// them.
+				for (GKInstance rleIE : newRLEInstanceEdits)
+				{
+					for (GKInstance pathwayIE : parentPathwayIEs)
+					{
+						if (rleIE.getDBID().equals(pathwayIE.getDBID()))
+						{
+							// The parent Pathway contained the IE.
+							hasRLEInstanceEdit = true;
+						}
+					}
+					if (hasRLEInstanceEdit)
+					{
+						// If it has the new IE, it might be not be the highest Pathway in the hierarchy
+						// with it.
+						// Continue looking up the hierarchy to try and find the HIGHEST Pathway with
+						// the IE.
+						GKInstance grandparentPathwayWithRLEInstanceEdit = getGrandparentPathwayWithRLEInstanceEdit(parentPathway, rleIE);
+						addPathwayToMap(pathwaysMap, newRLE, rleIE, grandparentPathwayWithRLEInstanceEdit);
+					}
+					else
+					{
+						addPathwayToMap(pathwaysMap, newRLE, rleIE, parentPathway);
+					}
+				}
+			}
 		}
-		catch (InvalidAttributeException e)
+		return pathwaysMap;
+	}
+
+	private static void addPathwayToMap(Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathwaysMap, GKInstance newRLE, GKInstance rleIE, GKInstance grandparentPathwayWithRLEInstanceEdit)
+	{
+		Map<GKInstance, GKInstance> reaction2newIE = new HashMap<>();
+		reaction2newIE.put(newRLE, rleIE);
+		// If 'Pathway' key exists, add to it
+		if (pathwaysMap.get(grandparentPathwayWithRLEInstanceEdit) != null)
 		{
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+			pathwaysMap.get(grandparentPathwayWithRLEInstanceEdit).add(reaction2newIE);
 		}
-		catch (Exception e)
+		else
 		{
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+			// If 'Pathway' key doesn't exist, create it and then add to it.
+			Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
+			pathwaysMap.put(grandparentPathwayWithRLEInstanceEdit, reaction2newIESet);
```

Comment:
This could be condensed by using the `computeIfAbsent` method:

`pathwaysMap.computeIfAbsent(grandparentPathwayWithRLEInstanceEdit, k -> new HashSet<>()).add(reaction2newIE);`

Reply:
Maybe in a future PR...